### PR TITLE
setup frontend deployments

### PR DESCRIPTION
**Pull request by Tayef-Shah**
## Notion ticket link

[Deploy Previews](https://www.notion.so/uwblueprintexecs/Deployment-Previews-e255c9574b44415eb5789e8f6b3b246b)

## Implementation description
* Followed [starter code guide](https://uwblueprint.github.io/starter-code-v2/docs/deployment/frontend-firebase-deployment.html) on frontend deployments with Firebase hosting
* Edited github actions to create client-side previews with each PR and a live preview from the development branch
* Added a couple of repo secrets for which backend URL the previews should access

## Steps to test
1. Click on the deployment preview

## What should reviewers focus on?
* Is development the branch we want live previews of or main?

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have run docker-compose up and my project compiled
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/.gitignore [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Logs
+logs
```
Comment (Tahmeed27) [inline frontend/.gitignore:2]: We didn't have a ,gitignore before :O?

**Review by Tahmeed27 — COMMENTED** [review]
So code-wise it seems fine and the preview url looks as expected. But I can't actually do anything like sign up because the preview url is rejected by CORS: 
image
There should be a way to allow all these preview urls to also interact with our backend right?

**Comment by Tahmeed27** [comment]
And to answer your comment - let's keep live previews for development for now

File: frontend/.gitignore (hunk 1 above)
Comment (Tayef-Shah) [reply frontend/.gitignore:2]: It was generated by the Firebase CLI, we can go back and remove unnecessary rules

**Comment by Tayef-Shah** [comment]
> So code-wise it seems fine and the preview url looks as expected. But I can't actually do anything like sign up because the preview url is rejected by CORS: image There should be a way to allow all these preview urls to also interact with our backend right?

Yes, we need to add a item to the cors allow list in the backend which will allow these previews to make requests to it.

**Comment by Tahmeed27** [comment]
> > So code-wise it seems fine and the preview url looks as expected. But I can't actually do anything like sign up because the preview url is rejected by CORS: image There should be a way to allow all these preview urls to also interact with our backend right?
> 
> Yes, we need to add an item to the cors allow list in the backend which will allow these previews to make requests to it.

Can you add that to the backend and re-deploy it so I can test that out before approving this PR?

**Comment by Tayef-Shah** [comment]
@Tahmeed27 I was just gonna write a regex to accept the deploy preview links as origins, but then I found out CORS doesn't allow regex. So I'm gonna have to find a new solution.

**Review by Tahmeed27 — APPROVED** [review]
LGTMM

**Pull request merged**